pkg/kubelet/status_manager_test.go

```diff
@@ -56,28 +56,71 @@ func verifyActions(t *testing.T, kubeClient client.Interface, expectedActions []
 	}
 }
 
+func verifyUpdates(t *testing.T, manager *statusManager, expectedUpdates int) {
+	// Consume all updates in the channel.
+	numUpdates := 0
+	for {
+		hasUpdate := true
+		select {
+		case <-manager.podStatusChannel:
+			numUpdates++
+		default:
+			hasUpdate = false
+		}
+
+		if !hasUpdate {
+			break
+		}
+	}
+
+	if numUpdates != expectedUpdates {
+		t.Errorf("unexpected number of updates %d, expected %s", numUpdates, expectedUpdates)
+	}
+}
+
 func TestNewStatus(t *testing.T) {
 	syncer := newTestStatusManager()
 	syncer.SetPodStatus(podFullName, getRandomPodStatus())
-	syncer.SyncBatch()
-	verifyActions(t, syncer.kubeClient, []string{"update-status-pod"})
+	verifyUpdates(t, syncer, 1)
 }
 
 func TestChangedStatus(t *testing.T) {
 	syncer := newTestStatusManager()
 	syncer.SetPodStatus(podFullName, getRandomPodStatus())
-	syncer.SyncBatch()
 	syncer.SetPodStatus(podFullName, getRandomPodStatus())
-	syncer.SyncBatch()
-	verifyActions(t, syncer.kubeClient, []string{"update-status-pod", "update-status-pod"})
+	verifyUpdates(t, syncer, 2)
 }
 
 func TestUnchangedStatus(t *testing.T) {
 	syncer := newTestStatusManager()
 	podStatus := getRandomPodStatus()
 	syncer.SetPodStatus(podFullName, podStatus)
-	syncer.SyncBatch()
 	syncer.SetPodStatus(podFullName, podStatus)
-	syncer.SyncBatch()
+	verifyUpdates(t, syncer, 1)
+}
+
+func TestSyncBatch(t *testing.T) {
+	syncer := newTestStatusManager()
+	syncer.SetPodStatus(podFullName, getRandomPodStatus())
+	err := syncer.syncBatch()
+	if err != nil {
+		t.Errorf("Unexpected syncing error: %v", err)
```

Comment:
Technically this is not an "error", but a log similar to glog.Errof() so in my head these are okay being upercase :) but the rest of the test disagrees so I made lowercase.